docs/proposals/promote-dependencies-automatically/README.md

```diff
@@ -0,0 +1,176 @@
+   2. If `Deps` is set to `true`, use the interpret framework to analyze the dependencies of the corresponding workload. Create the Resource Templates, ClusterPropagationPolicy/PropagationPolicy, and ResourceBindings for the dependencies in the Karmada control plane. Then, create the ResourceBinding for the workload itself. 
+
+**Note:**
+If the workload to be promoted is a CR(Custom Resource), we need to first promote its CRD (Custom Resource Definition).
+
+### Interpret workload dependencies
+
+To achieve the automatic promotion of dependent resources, the key challenge is to determine how to obtain the dependent resources of the resource to be promoted. Therefore, we need to be able to parse the dependencies of a resource. Currently, the interpret framework in Karmada supports the following methods for parsing dependent resources:
+
+1. Default built-in interpretation
+   1. Native Kubernetes Resource Interpretation:
+      1. Perform dependency parsing using the structure of native Kubernetes resources.
+   2. Third-Party Resource Interpretation:
+      1. Pre-configure Lua scripts for commonly used CR(Custom Resource) to enable dependency parsing using Lua configurations.
+3. Custom Interpretation
+   1. Lua Configuration-based Interpretation:
+      1. Configure resource interpretation rules for dependency parsing by specifying Lua scripts in the [ResourceInterpreterCustomization](https://github.com/karmada-io/karmada/blob/84b971a501ba82c53a5ad455c2fe84d842cd7d4e/pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go#L17) resource.
+   2. Webhook-based Interpretation:
+      1. Users can design Resource Interpreter Webhooks for specific custom resources to handle Interpreter Operations requests and perform dependency parsing. The process involves receiving these request operations and processing them accordingly.
+         The specific process is as follows:
+
+![webhook-interpret](statics/webhook-interpret.png)
+
+***Note:*** 
+Using the webhook-based interpretation method may pose difficulties when making requests to the webhook interpret service. In such cases, we need to perform port forwarding to invoke the service within the cluster.
+
+### Port forward the request to webhook interpret service
+
+Generally, external access to applications (Pods) running in Kubernetes can be achieved through three modes: NodePorts, LoadBalancer, and Ingress. Additionally, the `kubectl` subcommand `port-forward` can be used to forward ports and map local ports to specific Pod ports, allowing access to Pod services within the Kubernetes cluster.
+
+We can utilize the `port-forward` feature of `kubectl` to map a local port to the port of the webhook interpret Pod. This allows us to redirect webhook interpret requests initiated by `karmadactl` to the Pod's port and access the Pod service. The principle of `kubectl port-forward` is illustrated in the following diagram (Kubernetes v1.24.0):
```

Comment:
I think it's necessary to indicate that: 
We will implement the webhook requests based on the logic of `port-forward`, but not the native port-forward implementation.

our: data in program->create requests->send requests through the created port-forward connection->webhook pod.
native: data in program->create requests->send requests to port-forward listening address->send requests through the created port-forward connection->webhook pod.